naptime-testing/src/test/scala/org/coursera/naptime/NestedMacroCourierTests.scala

```diff
@@ -0,0 +1,107 @@
+package org.coursera.naptime
+
+import com.linkedin.data.schema.DataSchema
+import com.linkedin.data.schema.RecordDataSchema
+import org.coursera.naptime.couriertests.ExpectedMergedCourse
+import org.coursera.naptime.model.Keyed
+import org.coursera.naptime.resources.CourierCollectionResource
+import org.coursera.naptime.router2.Router
+import org.junit.Test
+import org.scalatest.junit.AssertionsForJUnit
+
+/**
+ * This test suite uses Courier to exercise advanced use cases for Naptime.
+ */
+object NestedMacroCourierTests {
+
+  class CoursesResource extends CourierCollectionResource[String, Course] {
+    override def resourceName: String = "courses"
+
+    override implicit lazy val Fields: Fields[Course] = BaseFields.withRelated("instructors" -> ResourceName("instructors", 1))
+
+    /**
+     * This `var` can be overridden to help fake out the implementation in particular functions.
+     */
+    var relatedFunction: Option[String => List[String]] = None
+
+    private[this] def fallbackRelatedFunction(id: String): List[String] = {
+      id match {
+        case "abc" => List("123")
+        case "xyz" => List.empty
+        case "qrs" => List("456", "789")
+        case _ => List.empty
+      }
+    }
+
+    private[this] def makeCourse(id: String): Keyed[String, Course] = {
+      val relatedInstructors = relatedFunction.getOrElse(fallbackRelatedFunction _)(id)
```

Comment:
oh, i forgot about the need to add that underscore in there to partially apply it. this looks uglier than i had hoped :/. sorry!